src/api/spec/models/user_ldap_strategy_spec.rb

```diff
@@ -189,4 +189,265 @@
       end
     end
   end
+
+  describe '#find_with_ldap' do
+    before do
+      stub_const('CONFIG', CONFIG.merge({
+        'ldap_search_user' => 'tux',
+        'ldap_search_auth' => 'tux_password',
+        'ldap_ssl' => :off,
+        'ldap_authenticate' => :ldap
+      }))
+    end
+
+    after do
+      # rspec-mocks doubles are not designed to last longer than for one
+      # example. Therefore we have to clear the stored connection.
+      UserLdapStrategy.class_variable_set(:@@ldap_search_con, nil)
+    end
+
+    context 'ldap doesnt connect' do
+      before do
+        allow(UserLdapStrategy).to receive(:initialize_ldap_con).and_return(nil)
+      end
+
+      subject! { UserLdapStrategy.find_with_ldap('tux', 'tux_password') }
+
+      it { is_expected.to be_nil }
+    end
+
+    context 'ldap connects' do
+      context 'ldap search works' do
+        include_context 'setup ldap mock'
+        include_context 'an ldap connection'
+
+        before do
+          allow(ldap_mock).to receive(:search)
+        end
+
+        subject! { UserLdapStrategy.find_with_ldap('tux', 'tux_password') }
+
+        it 'returns nil because the user was not found' do
+          is_expected.to be_nil
+        end
+      end
+
+      context 'without ldap_user_filter set' do
+        include_context 'setup ldap mock'
+        include_context 'an ldap connection'
+
+        before do
+          stub_const('CONFIG', CONFIG.merge({'ldap_user_filter' => nil }))
+
+          allow(ldap_mock).to receive(:search)
+        end
+
+        subject! { UserLdapStrategy.find_with_ldap('tux', 'tux_password') }
+
+        it 'returns nil because the user was not found' do
+          is_expected.to be_nil
+        end
+      end
+
+      context 'ldap search raises an error' do
+        include_context 'setup ldap mock'
+        include_context 'an ldap connection'
+
+        before do
+          allow(ldap_mock).to receive(:search).and_raise(ArgumentError)
+          allow(ldap_mock).to receive(:err).and_return('something went wrong')
+          allow(ldap_mock).to receive(:err2string).and_return('something went wrong')
+          allow(ldap_mock).to receive(:unbind)
+        end
+
+        subject! { UserLdapStrategy.find_with_ldap('tux', 'tux_password') }
+
+        it 'returns nil' do
+          is_expected.to be_nil
+        end
+      end
+
+      context 'ldap_authenticate = :local' do
```

Comment:
Independent from this PR: It might be worth to research if we can only have :ldap as ldap_authenticate. I am not sure if the other options are used and [our docu suggests not to use it either](http://openbuildservice.org/help/manuals/obs-admin-guide/obs.cha.administration.html#idm140000809067936):